miden-lib/src/notes/mod.rs

```diff
@@ -4,96 +4,104 @@ use miden_objects::{
     assets::Asset,
     notes::{Note, NoteScript},
     utils::{collections::Vec, vec},
-    Digest, Felt, Hasher, NoteError, Word, ZERO,
+    Felt, NoteError, Word, ZERO,
 };
 
+use miden_objects::crypto::rand::FeltRng;
+
 use super::transaction::TransactionKernel;
 
-// STANDARDIZED SCRIPTS
+mod utils;
+
+// STANDARDIZED SCRIPTS NOTE GENERATORS
 // ================================================================================================
 
-pub enum Script {
-    P2ID { target: AccountId },
-    P2IDR { target: AccountId, recall_height: u32 },
-    SWAP { asset: Asset, serial_num: Word },
+/// Generates a P2ID - pay to id note.
+pub fn create_p2id_note<R: FeltRng>(
+    sender: AccountId,
+    target: AccountId,
+    assets: Vec<Asset>,
+    mut rng: R,
+) -> Result<Note, NoteError> {
+    let assembler = TransactionKernel::assembler();
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2ID.masb"));
+    let script_ast = ProgramAst::from_bytes(bytes).map_err(NoteError::NoteDeserializationError)?;
+    let inputs = vec![target.into(), ZERO, ZERO, ZERO];
+    let (note_script, _) = NoteScript::new(script_ast, &assembler)?;
+    let tag: Felt = target.into();
+    let serial_num = rng.draw_word();
+    Note::new(note_script.clone(), &inputs, &assets, serial_num, sender, tag)
 }
 
-/// Users can create notes with a standard script. Atm we provide three standard scripts:
-/// 1. P2ID - pay to id.
-/// 2. P2IDR - pay to id with recall after a certain block height.
-/// 3. SWAP - swap of assets between two accounts.
-pub fn create_note(
-    script: Script,
-    assets: Vec<Asset>,
+/// Generates a P2IDR - pay to id with recall after a certain block height.
+pub fn create_p2idr_note<R: FeltRng>(
     sender: AccountId,
-    tag: Option<Felt>,
-    serial_num: Word,
+    target: AccountId,
+    assets: Vec<Asset>,
+    recall_height: u32,
+    mut rng: R,
 ) -> Result<Note, NoteError> {
-    let note_assembler = TransactionKernel::assembler();
+    let assembler = TransactionKernel::assembler();
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2IDR.masb"));
+    let script_ast = ProgramAst::from_bytes(bytes).map_err(NoteError::NoteDeserializationError)?;
+    let inputs = vec![target.into(), recall_height.into(), ZERO, ZERO];
+    let (note_script, _) = NoteScript::new(script_ast, &assembler)?;
+    let tag: Felt = target.into();
+    let serial_num = rng.draw_word();
+    Note::new(note_script.clone(), &inputs, &assets, serial_num, sender, tag)
+}
+
+/// Generates a SWAP - swap of assets between two accounts.
+pub fn create_swap_note<R: FeltRng>(
+    sender: AccountId,
+    offered_asset: Asset,
+    requested_asset: Asset,
+    mut rng: R,
+) -> Result<(Note, Note), NoteError> {
```

Comment:
For now, let's change the return type to `Result<(Note, Word), NoteError>` and return the internally generated serial number as the second member of the tuple.

Once we refactor the `Note` struct, we'll come back here and replace the `Word` with `Note`.